packages/codegen-ui-react/lib/__tests__/studio-ui-codegen-react.test.ts

```diff
@@ -128,6 +129,25 @@ describe('amplify render tests', () => {
       const generatedCode = generateWithAmplifyRenderer('collectionWithBindingItemsName');
       expect(generatedCode.componentText).toMatchSnapshot();
     });
+
+    it('should render nested query if model has a hasMany relationship', () => {
+      const { componentText } = generateWithAmplifyRenderer(
+        'authorCollectionComponent',
+        {},
+        false,
+        authorHasManySchema,
+      );
+      expect(componentText).toContain('const bookItems = useDataStoreBinding');
+      expect(componentText).toContain('const itemsDataStore = useDataStoreBinding');
+      expect(componentText).toMatchSnapshot();
```

Comment:
move snapshots to dedicated tests to avoid making these unit tests brittle

Reply:
keeping this as a snapshot test file will move to another test file to test more so on the componentText itself